### PR TITLE
Add optional password field on user creation

### PR DESCRIPTION
**Pull request by dkanada**

### CODE REVIEW

File: src/controllers/usernew.js [hunk 1]
```diff
@@ -3,74 +3,95 @@ define(["jQuery", "loading", "fnchecked", "emby-checkbox"], function($, loading)
...
+        html += '<div class="checkboxList paperList" style="padding:.5em 1em;">';
+        for (var i = 0; i < channels.length; i++) {
             var folder = channels[i];
-            html += '<label><input type="checkbox" is="emby-checkbox" class="chkChannel" data-id="' + folder.Id + '" checked="checked"/><span>' + folder.Name + "</span></label>"
+            html += '<label><input type="checkbox" is="emby-checkbox" class="chkChannel" data-id="' + folder.Id + '" checked="checked"/><span>' + folder.Name + "</span></label>";
         }
-        html += "</div>", $(".channelAccess", page).show().html(html).trigger("create"), channels.length ? $(".channelAccessContainer", page).show() : $(".channelAccessContainer", page).hide(), $("#chkEnableAllChannels", page).checked(!0).trigger("change")
+        html += "</div>";
+        $(".channelAccess", page).show().html(html).trigger("create");
+        channels.length ? $(".channelAccessContainer", page).show() : $(".channelAccessContainer", page).hide();
```
Comment (cvium) [inline src/controllers/usernew.js]:
```suggestion
        if (channels.length) {
            $(".channelAccessContainer", page).show();
        } else {
            $(".channelAccessContainer", page).hide();
        }
```

File: src/controllers/usernew.js [hunk 2]
```diff
@@ -3,74 +3,95 @@ define(["jQuery", "loading", "fnchecked", "emby-checkbox"], function($, loading)
...
     function saveUser(page) {
-        var name = $("#txtUserName", page).val();
-        ApiClient.createUser(name).then(function(user) {
-            user.Policy.EnableAllFolders = $("#chkEnableAllFolders", page).checked(), user.Policy.EnabledFolders = user.Policy.EnableAllFolders ? [] : $(".chkFolder", page).get().filter(function(i) {
+        var user = {};
+        user.Name = $("#txtUsername", page).val();
+        user.Password = $("#txtPassword", page).val();
+        ApiClient.createUser(user).then(function(user) {
+            user.Policy.EnableAllFolders = $("#chkEnableAllFolders", page).checked();
+            user.Policy.EnabledFolders = user.Policy.EnableAllFolders ? [] : $(".chkFolder", page).get().filter(function(i) {
```
Comment (cvium) [inline src/controllers/usernew.js]: I don't like these ternaries. They should be deminified as well imo.

File: src/controllers/usernew.js [hunk 3]
```diff
@@ -3,74 +3,95 @@ define(["jQuery", "loading", "fnchecked", "emby-checkbox"], function($, loading)
...
-            }), ApiClient.updateUserPolicy(user.Id, user.Policy).then(function() {
-                Dashboard.navigate("useredit.html?userId=" + user.Id)
-            })
+                return i.getAttribute("data-id");
+            });
+            ApiClient.updateUserPolicy(user.Id, user.Policy).then(function() {
+                Dashboard.navigate("useredit.html?userId=" + user.Id);
+            });
         }, function(response) {
-            400 == response.status ? Dashboard.alert({
```
Comment (cvium) [inline src/controllers/usernew.js]: Why was this removed?

File: src/controllers/usernew.js [hunk 4]
```diff
@@ -3,74 +3,95 @@ define(["jQuery", "loading", "fnchecked", "emby-checkbox"], function($, loading)
...
     }
+
     $(document).on("pageinit", "#newUserPage", function() {
         var page = this;
         $("#chkEnableAllChannels", page).on("change", function() {
-            this.checked ? $(".channelAccessListContainer", page).hide() : $(".channelAccessListContainer", page).show()
-        }), $("#chkEnableAllFolders", page).on("change", function() {
-            this.checked ? $(".folderAccessListContainer", page).hide() : $(".folderAccessListContainer", page).show()
-        }), $(".newUserProfileForm").off("submit", onSubmit).on("submit", onSubmit)
+            this.checked ? $(".channelAccessListContainer", page).hide() : $(".channelAccessListContainer", page).show();
```
Comment (cvium) [inline src/controllers/usernew.js]: Same here

File: src/controllers/usernew.js (hunk 3 above)
Comment (dkanada) [reply src/controllers/usernew.js]: The element it queried doesn't exist.

File: src/controllers/usernew.js [hunk 5]
```diff
@@ -49,17 +53,23 @@ define(["jQuery", "loading", "fnchecked", "emby-checkbox"], function($, loading)
...
         ApiClient.createUser(user).then(function(user) {
             user.Policy.EnableAllFolders = $("#chkEnableAllFolders", page).checked();
-            user.Policy.EnabledFolders = user.Policy.EnableAllFolders ? [] : $(".chkFolder", page).get().filter(function(i) {
-                return i.checked
-            }).map(function(i) {
-                return i.getAttribute("data-id");
-            });
+            user.Policy.EnabledFolders = [];
+            if (!user.Policy.EnableAllFolders) {
+                $(".chkFolder", page).get().filter(function(i) {
```
Comment (cvium) [inline src/controllers/usernew.js]: Forgot the assignment

File: src/controllers/usernew.js [hunk 6]
```diff
@@ -49,17 +53,23 @@ define(["jQuery", "loading", "fnchecked", "emby-checkbox"], function($, loading)
...
+            }
             user.Policy.EnableAllChannels = $("#chkEnableAllChannels", page).checked();
-            user.Policy.EnabledChannels = user.Policy.EnableAllChannels ? [] : $(".chkChannel", page).get().filter(function(i) {
-                return i.checked
-            }).map(function(i) {
-                return i.getAttribute("data-id");
-            });
+            user.Policy.EnabledChannels = [];
+            if (!user.Policy.EnableAllChannels) {
+                $(".chkChannel", page).get().filter(function(i) {
```
Comment (cvium) [inline src/controllers/usernew.js]: Same
